lib/modules/dosomething/dosomething_signup/includes/SignupsController.php

```diff
@@ -2,6 +2,8 @@
 
 class SignupsController extends EntityAPIController {
 
+  protected $ignore_global_user = FALSE;
```

Comment:
Oh do you mean updating the method to not require the `global $user` and only use the passed in one?